services/sns.cfc

```diff
@@ -139,6 +137,29 @@ component {
             if ( len( arguments[ key ] ) ) formParams[ key ] = arguments[ key ];
         }
 
+        if ( arguments.keyExists( "MessageAttributes" ) ) {
+            var i = 1;
+            MessageAttributes.each(function(k, v) {
+                var dataType = "String";
+
+                if ( isNumeric(v) ) {
+                    dataType = "Number";
+                } else if ( isArray(v) ) {
+                    dataType = "String.Array";
+                }
+
+                formParams [ "MessageAttributes.entry.#i#.Name" ] = k;
+                formParams [ "MessageAttributes.entry.#i#.Value.DataType" ] = dataType;
+                if ( dataType == "String.Array" ) {
+                    formParams [ "MessageAttributes.entry.#i#.Value.StringValue" ] = "[\""#v.toList("\"", \""")#\""]";
```

Comment:
Using `\"` to escape quotes is not a CFML thing. 😄 Also, I took a look at the docs and it looks like `String.Array` can include strings, numbers, and the keywords `true`, `false`, and `null`. Of those, only the strings need to be quoted. So I actually think this line could just use `serializeJSON(v)` to format the array correctly.

Reply:
I was directly copying one of the AWS Java examples for this portion, definitely not idiomatic coldfusion 😬 thanks